### ISSUE TITLE
[Code Quality] Replace Magic Numbers with Named Constants and Configuration

### ISSUE THREAD

**Issue by BenSitn** [H-EmbodVis/GRANT#3]
The codebase contains numerous "magic numbers" and hardcoded thresholds scattered throughout models, datasets, and trainer code. These unexplained literal values make the code difficult to understand, maintain, and tune. Moving these values to named constants or configuration files would improve code clarity and enable easier experimentation.

## Problem Statement

Magic numbers reduce code quality in several ways:
1. **Unclear Intent**: Values like `0.5`, `200`, `1e-4` lack context about their purpose
2. **Difficult Tuning**: Changing hyperparameters requires searching through code
3. **Inconsistency**: Same conceptual values may differ across files
4. **Error-Prone**: Easy to accidentally modify when editing nearby code
5. **Poor Documentation**: No explanation of why specific values were chosen

#### 1. Dataset Configuration (datasets/sceneverse.py)

**Lines 21-24:**
```python
self.max_obj_len = 200        # ❌ Magic number - max objects per scene
self.max_scene_len = 2048     # ❌ Magic number - max points per scene
self.max_lang_len = 256       # ❌ Magic number - max language tokens
self.num_points = 50000       # ❌ Magic number - point cloud sampling
```

**Lines 156-158:**
```python
if len(point_clouds) > 100000:    # ❌ Hardcoded threshold
    point_clouds = point_clouds[:100000]

if len(lang_tokens) > 512:        # ❌ Different from max_lang_len above
    lang_tokens = lang_tokens[:512]
```

#### 2. Model Dimensions (models/LLM/LLama3d.py)

**Lines 47-52:**
```python
self.hidden_dim = 4096           # ❌ Magic number
self.num_heads = 32              # ❌ Magic number
self.num_layers = 32             # ❌ Magic number
self.intermediate_size = 11008   # ❌ Magic number - why 11008?
self.max_position_embeddings = 2048  # ❌ Magic number
self.rope_theta = 10000.0        # ❌ Magic number for RoPE encoding
```

#### 3. Point Cloud Encoder (models/point_encoder.py)

**Lines 89-92:**
```python
self.voxel_size = 0.02           # ❌ Magic number - 2cm voxels
self.min_points_per_voxel = 5    # ❌ Magic number
self.radius = 0.1                # ❌ Magic number - 10cm radius
self.num_neighbors = 16          # ❌ Magic number - K-NN neighbors
```

#### 4. Trainer Configuration (trainer/trainer.py)

**Lines 123-128:**
```python
self.learning_rate = 1e-4        # ❌ Magic number
self.weight_decay = 0.01         # ❌ Magic number
self.warmup_steps = 1000         # ❌ Magic number
self.max_grad_norm = 1.0         # ❌ Magic number
self.label_smoothing = 0.1       # ❌ Magic number
self.dropout_rate = 0.1          # ❌ Magic number
```

**Lines 234-238:**
```python
if epoch < 10:                   # ❌ Magic number - when to enable augmentation
    use_augmentation = False
else:
    use_augmentation = True

loss_weight = 0.5 if task == 'grounding' else 1.0  # ❌ Magic weights
```

#### 5. Loss Computation (trainer/trainer.py)

**Lines 445-450:**
```python
ce_loss = F.cross_entropy(logits, targets, label_smoothing=0.1)  # ❌ Duplicate value
bbox_loss = F.l1_loss(pred_boxes, target_boxes) * 5.0           # ❌ Magic weight
iou_loss = self.compute_iou_loss(pred_boxes, target_boxes) * 2.0  # ❌ Magic weight

total_loss = ce_loss + bbox_loss + iou_loss

if confidence_score < 0.3:       # ❌ Magic threshold
    total_loss *= 0.5            # ❌ Magic dampening factor
```

#### 6. Data Augmentation (datasets/utils.py)

**Lines 67-75:**
```python
# Random rotation
max_rotation = 15.0              # ❌ Magic number - degrees
rotation_angle = np.random.uniform(-max_rotation, max_rotation)

# Random scaling
scale_factor = np.random.uniform(0.8, 1.2)  # ❌ Magic range

# Random jitter
jitter_std = 0.01                # ❌ Magic number - 1cm std dev
points += np.random.normal(0, jitter_std, points.shape)

# Color augmentation
if np.random.rand() > 0.5:       # ❌ Magic probability
    colors *= np.random.uniform(0.7, 1.3, (1, 3))  # ❌ Magic range
```

#### 7. Filtering and Validation (datasets/preprocessing/filter.py)

**Lines 123-130:**
```python
if num_objects < 3:              # ❌ Magic minimum
    return False  # Skip scenes with too few objects

if num_points < 5000:            # ❌ Magic minimum
    return False  # Skip sparse scenes

if language_length < 10:         # ❌ Magic minimum
    return False  # Skip short descriptions

if iou_overlap > 0.8:            # ❌ Magic threshold
    return False  # Skip overlapping objects
```

#### 8. Prediction Thresholds (models/metrics/utils.py)

**Lines 89-95:**
```python
# Confidence threshold
predictions = predictions[predictions[:, -1] > 0.5]  # ❌ Magic threshold

# Non-maximum suppression
keep_indices = nms(boxes, scores, iou_threshold=0.3)  # ❌ Magic IoU

# Top-k selection
predictions = predictions[:50]    # ❌ Magic number - top 50

# Distance threshold for grounding
distance_threshold = 0.5          # ❌ Magic number - 50cm
```

#### 9. Scheduling Parameters (models/scheduler.py)

**Lines 45-52:**
```python
# Task scheduling thresholds
min_duration = 5.0               # ❌ Magic number - 5 seconds minimum
max_parallel_tasks = 3           # ❌ Magic number
time_buffer = 2.0                # ❌ Magic number - 2 second buffer

# Priority weights
priority_weights = {
    'urgent': 2.0,               # ❌ Magic weights
    'normal': 1.0,
    'low': 0.5
}
```

#### 10. Checkpoint Management (utils/checkpoint.py)

**Lines 67-72:**
```python
# Keep only top-k checkpoints
max_checkpoints_to_keep = 5      # ❌ Magic number

# Save interval
save_every_n_epochs = 10         # ❌ Magic number

# Checkpoint size validation
if checkpoint_size_mb > 5000:    # ❌ Magic number - 5GB limit
    logger.warning("Checkpoint too large!")
```

**Comment by zc2023** [comment]
Thank you for your suggestion. These values are commonly understood by experts in the field.
